src/js/client/ajaxCapable.js

```diff
@@ -0,0 +1,142 @@
+/*
+
+  A component to standardize the process of retrieving data and then updating the model when the results are received.
```

Comment:
We need to leave a note that this needs to be unified with the framework's dataSource facility (both in infusion and kettle) once this is complete. It should be possible to operate the same semantics against any source of data, not just an AJAX request